modules/builder/src/builders/helpers/xviz-trajectory-helper.js

```diff
@@ -117,7 +118,7 @@ export function getObjectTrajectory({
     // objects in curr frame are meters offset based on current vehicle pose
     // need to convert to the coordinate system of the start vehicle pose
     const p = transformMatrix.transformVector([step.x, step.y, step.z]);
-    vertices.push([p[0] + x, p[1] + y, p[2]]);
+    vertices.push([p[0] + x, p[1] + y, p[2] + z]);
```

Comment:
@Pessimistress 

As we discussed earlier, you mentioned `[x, y, z]` from `getGeospatialVector` can not be added to `p` directly.

Maybe I did not understand correctly, but in the code, `p` is transformed to `startVehiclePose ` coordinate system, and so does `[x, y, z]` result from `getGeospatialVector`. 

`const [x, y, z] = getGeospatialVector(startVehiclePose, currVehiclePose, startVehiclePose.yaw);` calculate `x` and `y` values based on `startVehiclePose.yaw`.

Reply:
All of roll, pitch, and yaw affect `p`. Only adding `yaw` does not make xyz in the same coordinate system as the `startVehiclePose`.